manage-mysql-user.py

```diff
@@ -6,66 +6,91 @@
 import boto3
 import random
 import string
-
-from aws_xray_sdk.core import xray_recorder
-from aws_xray_sdk.core import patch
+import json
 
 libraries = ("boto3", "mysql")
-patch(libraries)
 
 # Initialise logging
 logger = logging.getLogger(__name__)
 log_level = os.environ["LOG_LEVEL"] if "LOG_LEVEL" in os.environ else "INFO"
 logger.setLevel(logging.getLevelName(log_level.upper()))
 logger.info("Logging at {} level".format(log_level.upper()))
 
+# Use appropriate CA SSL cert to verify RDS identity. Use "AmazonRootCA1.pem" for Aurora Serverless.
+# Defaults to "rds-ca-2019-2015-root.pem" to suit normal RDS.
+if "RDS_CA_CERT" in os.environ:
+    rds_ca_cert = os.environ["RDS_CA_CERT"]
+else:
+    rds_ca_cert = "/var/task/rds-ca-2019-2015-root.pem"
+
 
 def generate_password():
     valid_chars = string.ascii_letters + string.digits + string.punctuation
     invalid_chars = ["/", "@", '"', "\\", "'"]  # Not allowed in a MySQL password
     pw_chars = "".join([i for i in valid_chars if i not in invalid_chars])
     pw = "".join((random.choice(pw_chars)) for x in range(40))
+    logger.debug("Password generated")
     return pw
 
 
-def update_password_parameter(password, parameter_name):
-    ssm = boto3.client("ssm")
-    try:
-        ssm.put_parameter(
-            Name=parameter_name, Type="SecureString", Value=password, Overwrite=True
-        )
-    except Exception as e:
-        logger.error(e)
-
-
-def get_master_password():
-    ssm = boto3.client("ssm")
-    return ssm.get_parameter(
-        Name=os.environ["RDS_MASTER_PASSWORD_PARAMETER_NAME"], WithDecryption=True
-    )["Parameter"]["Value"]
+def update_password_source(password, password_source, password_source_type):
```

Comment:
I found the naming a little confusing. Both parameter store and secrets manager refer to the `password_source` as a `name`, so maybe something like:
```suggestion
def update_password_source(password_name, password_value, password_store):
```

Reply:
It is all a bit confusing, not least because we are dealing with passwords and things containing passwords and doing it multiple times - for master user and the user we are updating. 
`password_source` is always the name of AWS entity storing a password, e.g. an SSM parameter or Secrets Manager secret. 
`password` is the actual password value generated by `generate_password`, so I think the name is right.